django/template/loaders/filesystem.py

```diff
@@ -16,8 +16,12 @@
 
 class Loader(BaseLoader):
 
+    def __init__(self, engine, dirs=None):
+        super(Loader, self).__init__(engine)
+        self.dirs = dirs
+
     def get_dirs(self):
-        return self.engine.dirs
+        return self.dirs or self.engine.dirs
```

Comment:
I'd go with if `return self.dirs if self.dirs is not None else self.engine.dirs` -- it's possible that someone might instantiate the loader with an empty list (say in a debugging scenario if they are temporarily removing the one entry in the list to test something), and it's surprising to fall back to engine-dirs in that case.